gui/elements/qt/_layer.py

```diff
@@ -0,0 +1,52 @@
+from PyQt5.QtWidgets import QSlider, QComboBox, QHBoxLayout, QGroupBox, QVBoxLayout, QCheckBox
+from PyQt5.QtCore import Qt
+
+class QtLayer(QGroupBox):
+    def __init__(self, layer):
+        super().__init__(layer.name)
+        self.layer = layer
+        layout = QHBoxLayout()
+
+        cb = QCheckBox('Visibility', self)
+        cb.setChecked(self.layer.visible)
+        cb.stateChanged.connect(lambda state=cb: self.changeVisible(state))
+        layout.addWidget(cb)
+
+        comboBox = QComboBox()
+        for cmap in self.layer.colormaps:
```

Comment:
This is specific to `Image` layers. I'd suggest renaming the `_qt` property on layers to something like `_qt_panel` and creating a `_qt_layer_entry` property that inserts itself into the layer entry box here.

Reply:
I'm starting to think about this issue right now, but not sure I fully grasp it. I think it makes sense to use the `_qt` property that gets defined in the `_base` file defining the `Layer` class. Right now this just gets set to return `None`, but instead I want to be able to set it as a property when either and `image` or `marker` layer is being created. At that time the appropriate widget can be imported and attached. Attaching the widget on the layer like this will also make it easier to rearrange the widgets when the `LayersList` gets rearranged or edited too. Does this make sense? Do you have advice on how to make the `_qt` property settable?

Reply:
I've given it a go and pushed the changes, will comment more in the main thread